### PR TITLE
[#10119] Android app UI is slow after  background

### LINKED ISSUE #10119: Android app UI is slow when recieving new messages
# Bug Report
## Problem
When i open app UI is slow. Sometimes it's slow during using it

#### Expected behavior
UI is smooth

## Investigations 
_Updated 5 Mar 2020_

* [x] 1) `dispatch-later` is paused when app in background, use `:utils/dispatch-later` instead
* [ ] 2) it seems after a long time in background `:utils/dispatch-later` works differently as well (not proved technically, only visually) 
* [x] 3) `dispatch-later` with 20ms leads to subs and views recalculations, if dispatch is used, subs recalculated only once , `dispatch-later` should be replaced by another approach 
* [ ] 4) parametrized subscriptions leads to a lot of computations when app-db is changed, do not use them
* [ ] 5) subscriptions usage is not optimized, shouldn't be used in chat views at all, currently, we recalculate all data for every message, message should be added in event, subscription must only return the result without recalculating data
* [ ] 6) `process-response` signal is to often, probably we should send it not more than once a second
* [ ] 7) move all data modification to status-go, currently, we recalculate timestamps for all messages in subscription when receiving every new message
* [x] 8) `cljs-bean` should be used only when you want to access js object , not when you want to modify this data
* [ ] 9) in `flat-list` new array is created each time data is changed 

## ideal implementation
status-go signals json with new data -> status-react converts json to js, and append data to js array , js array is set to flat-list directly, no cljs data or subscriptions involved

### PR DESCRIPTION
**Pull request by flexsurfer**
partially f i x e s  #10119

### CODE REVIEW

File: src/status_im/contact/core.cljs [hunk 1]
```diff
@@ -60,19 +60,9 @@
...
    {:keys [public-key] :as contact}]
-  (let [new? (get-in db [:contacts/contacts public-key])
-        us? (= public-key (multiaccounts.model/current-public-key cofx))]
-    (fx/merge cofx
-              {:db (-> db
-                       (update-in [:contacts/contacts public-key] merge contact))}
-
-              (cond
-                us?
-                (handle-update-from-contact-request contact)
```
Comment (rasom) [inline src/status_im/contact/core.cljs:73]: this code was doing nothing important?

Comment (flexsurfer) [reply src/status_im/contact/core.cljs:73]: nothing :) но это не точно :)

Comment (rasom) [reply src/status_im/contact/core.cljs:73]: oh nice, that's very comforting! :)

**Comment by rasom** [comment]
If that's because of `:dispatch-later` queuing, can we try `:utils/dispatch-later` instead?

**Comment by flexsurfer** [comment]
@Serhy could you please test this one when joining public chat with thousands of messages

**Comment by Serhy** [comment]
Something is wrong. 
With latest https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-200303-110209-40948b-pr-universal.apk / https://i.diawi.com/pEgCzY builds after new account is created/recovered there is endless spinner instead `(+)` button.
Fresh build installed.

After I re-open app and try to loging then a `sql: no rows in result set` error appears.

These are the logs after account created:
[loagcat.log](https://github.com/status-im/status-react/files/4281621/dispach-later-mod.log)

**Comment by flexsurfer** [comment]
@Serhy probably it's latest develop, not this PR :(

**Comment by churik** [comment]
@flexsurfer can confirm https://github.com/status-im/status-react/pull/10120#issuecomment-593972666 - endless spinner with builds from this PR and works normally in latest develop (nightly 04/03/2020)

**Comment by flexsurfer** [comment]
@Serhy @churik thanks, fixed

**Comment by Serhy** [comment]
@flexsurfer I tested this build on Galaxy Note 4 and doesn't look good (it got the similar behavior to what we had before moving processing of messages into status-go)

So with https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-200303-110209-40948b-pr-universal.apk build when I join public chat having 1000+ messages during last 24h (stickers in my case), - then app freezes while all messages are coming up to 40-43sec (video https://drive.google.com/file/d/15jLN7ElNjHWicFpo9tubsXDg51jDP76b/view?usp=sharing)

While there are no lags at all when joining the same public chat in nightly (4th March) build.

**Comment by flexsurfer** [comment]
thanks @Serhy , ok so i reverted this changes, and now this PR only fixes performance issue when open app from background

how to repro
1) hide app in background,
2) send lots of message to this user or public chat
3) open from background after  5-10m

**Review by rasom — APPROVED** [review]

**Comment by Serhy** [comment]
No bugs found in latest build: opening app from background performance issue not reproduced, https://github.com/status-im/status-react/pull/10120#issuecomment-594455979 (when moving between screens also not reproduced).
With @churik help we did not see any regressions

Lets merge, @flexsurfer

**Pull request merged**